scalanet/src/io/iohk/scalanet/peergroup/dynamictls/DynamicTLSPeerGroupInternals.scala

```diff
@@ -281,9 +277,9 @@ private[dynamictls] object DynamicTLSPeerGroupInternals {
 
     override def close(): Task[Unit] =
       for {
-        _ <- Task.now(log.debug("Closing client channel to peer {}", to))
-        _ <- Task(messageSubject.onComplete())
```

Comment:
Why is this no longer needed?

Reply:
now `messageSubject.onComplete` is called only in `MessageNotifier.channelInactive` handler. This approach removes any possible violation of subject contract as all calls to message subject i.e`onNext`, `onComplete`, `onError` ale always happening on the same netty thread.